internal/controllers/addon/controller.go

```diff
@@ -53,6 +55,8 @@ type AddonReconciler struct {
 
 	ocmClient    ocmClient
 	ocmClientMux sync.RWMutex
+	// reconciler Fn
+	reconciler reconcilerFn
```

Comment:
this is very clever. I appreciate the thought went into this refactoring to make it easily mockable for tests. However, for a future developer reading this code, this could look confusing. This struct has `reconciler` and then `subReconcilers`. Im okay with merging it as it is. However, i have 2 suggestions to consider:

1. leave this pr as it is but add a longer comment here explaining that this field is here to aid with testing.
2. rename the `subReconcilers` filed to `reconcilers`. then capture logic from lines 308 to 350 in a new recociler (sub) 

3. 👨‍💻👉 revert this change in this struct. while testing add a sub-reconciler to the `AddonReconciler` struct if `reconcilerErrPresent` is `true`

Reply:
Yeah i didn't think of just mocking the sub-reconciler for returning errors from the main reconcile method. (3am mushy brain i guess xD). Thanks for the suggestion. :)